C_CPP/Classes & Objects Programming/Polymorphism.md

```diff
@@ -0,0 +1,149 @@
+  public:
+  
+  // 1st function with number of arguments 1 and signature of int type.
+  
+  void Function(int a){   
+   cout<<"\nYour Entered Integer is : "<<a;
+  }
+  
+  // function with same name with number of arguments 1 and with different signature of float.
+  
+  void Function(double a){ 
+      cout<<"\nYour Entered Float number is :"<<a;
+  }
+  
+  // function with same name and with number of arguments 2 and with signature same as 1st function.
+  void Function(int a,int b){ //
+      cout<<"\n Your Entered Two integers are :"<<a<<" , "<<b;
+  }
+};
+    
+int main() {
+	Function_Overloading fo;
+	
+	// calling the first function
+	fo.Function(5);
+	
+	// calling the second function
+	fo.Function(3.4);
+	
+	//calling the third Function
+	fo.Function(5,6);
+	return 0;
+}
+```
+Output :
+```
+Your Entered Integer is : 5
+Your Entered Float number is : 3.4
+ Your Entered Two integers are : 5 , 6
+```
+
+### 2. Operator Overloading
+<p> If we talk about operator overloading in this we overload the operators eg. ' + ' , ' - ' etc , if take an example, we can concatenate two strings,for this task we can use the single operator ' + ' which can concatenate the two strings with two strings as as operands.</p>
+
+```
+#include <iostream>
+using namespace std;
+class Complex_addition {
+private:
+    int real,imaginary;
+public:
+    Complex_addition(int r=0,int i=0)  
+    {real = r;   
+    imaginary = i;}
+    // addition between two Complex numbers and Overloading the operator
+    Complex_addition operator + (Complex_addition &c) {
+         Complex_addition com;
+         com.real = real + c.real;
+         com.imaginary = imaginary + c.imaginary;
+         return com;
+    }
+    void print_complex() 
+    { cout << real << " + " << imaginary<<"i"<<"\n"; 
+        
+    }
+};
+   
+int main()
+{
+    Complex_addition com1(2, 3);
+    Complex_addition com2(4, 5);
+    Complex_addition com3 = com1 + com2; // calling the operator overloaded function "operator+"
+    com3.print_complex();
+}
+```
+Output :
+```
+6 + 8i
+
+```
+
+## --> Run Time polymorphism
+<p>This is the polymorphism which occur at run time and this can be archieved by function overriding.     </p>
+
+### 1. Function Overridding
+<p>This is occur commonly in classes when derived and parent class both have the member fuction of same name in that one of the member function of the parent class hide the derive class member function so to avoid this we use " virtual " keyword which heps us to tackle this problem.</p>
+
+```
+#include <iostream>
+using namespace std;
+
+class parent
+{
+public:
+    virtual void display()
+    { cout<< "this is our base class" <<"\n"; }
+   
+    void show ()
+    { cout<< "showing the base class" <<"\n"; }
+};
+   
+class child:public parent
+{
+public:
+//display() is already virtual function in derived class and we also declared as virtual void display() and it will be called when we call display function.
+    void display() 
```

Comment:
Add one line space before and after you add a comment.

Reply:
Sir i have done all the needful changes please review it again.

Reply:
### 1. indentation
### 2. Space between the comments.
Issue number closed : #835